src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -124,53 +124,61 @@ SymbolKind.NetModule or
                     ForkState(oldProjectState: this.ProjectState, ReadState(), translate),
                     this.SkeletonReferenceCache.Clone());
 
-                static CompilationTrackerState? ForkState(
+                static InProgressState? ForkState(
                     ProjectState oldProjectState,
                     CompilationTrackerState? state,
                     CompilationAndGeneratorDriverTranslationAction? translate)
                 {
                     if (state is null)
                         return null;
 
+                    // Determine the old/stale compilation to help seed the non-final state with.
+                    var staleCompilationWithGeneratedDocuments = state switch
+                    {
+                        FinalState finalState => finalState.FinalCompilationWithGeneratedDocuments,
+                        InProgressState inProgressState => inProgressState.StaleCompilationWithGeneratedDocuments,
```

Comment:
note: i know you mentioned moving StaleCompilationWithGeneratedDocuments to the root class @jasonmalinowski .  but i still don't like that :)